tests/semanticdb/expect/RecOrRefined.scala

```diff
@@ -0,0 +1,18 @@
+package example
+
+def m1(a: Int { val x: Int }) = ???
+def m2(x: { val x: Int; def y: Int }) = ???
+def m3(x: { val x: Int; def y: Int; type z }) = ???
```

Comment:
could be good to test a polymorphic refinement, which is allowed if it is not structural:
```scala
trait PolyHolder {
  def foo[T](t: T): Any
}

...

def m4(x: PolyHolder { def foo[T](t: T): T }) = ???
```

Reply:
btw, the refinements are printed as empty if the refinement type is the result of a polymorphic method:
```scala
def m5[Z](x: Int): PolyHolder { def foo[T](t: T): T } = ???
```

Reply:
Fixed https://github.com/lampepfl/dotty/pull/12885/commits/e68287a1229928ba27cae88e3f88d64ed30e0b70